nearswap/src/lib.rs

```diff
@@ -1160,34 +1143,21 @@ mod tests {
         c.set_pool(&t2, &p2);
 
         let amount: u128 = 1_000_000;
-        let amount_net: u128 = 997_000;
         let mut v = c.price_near_to_token_in(t1.clone(), amount.into());
-        let v_expected = amount_net * p1.tokens / (p1.ynear + amount_net);
+
+        let mut v_expected = mock_calc_price_fee(amount, p1.ynear, p1.tokens);
         assert_eq!(to_num(v), v_expected);
-        let mut v2 = c.price_near_to_token_out(t1.clone(), v.into());
-        assert_eq!(to_num(v2), amount, "price_out(price_in) must be identity");
 
         // check reverse computation
-        v = c.price_token_to_near_in(t1.clone(), (amount * p1_factor).into());
-        assert_eq!(to_num(v), v_expected / p1_factor);
-        v2 = c.price_token_to_near_out(t1.clone(), v.into());
-        assert_close(v2, amount * p1_factor, 10);
-
-        // we can also do a reverse computation by starting from expected output
-        v2 = c.price_token_to_near_out(t1.clone(), v_expected.into());
-        assert_close(v2, amount * p1_factor * p1_factor, 10);
-        v = c.price_token_to_near_in(t1.clone(), v2.into()).into();
+        v_expected = mock_calc_price_fee(amount, p1.tokens, p1.ynear);
+        v = c.price_token_to_near_in(t1.clone(), amount.into());
         assert_eq!(to_num(v), v_expected);
 
         /*
          * test  token1 -> token2 swap
          */
         v = c.price_token_to_token_in(t1.clone(), t2.clone(), amount.into());
-        v2 = c.price_token_to_token_out(t1.clone(), t2.clone(), v.into());
-        assert_close(v2, amount, 10);
-
-        v2 = c.price_token_to_token_out(t2.clone(), t1.clone(), amount.into());
-        assert_close(v2, v.into(), amount * 2 / 1_000);
+        assert_close(v, amount / 8, 1000);
```

Comment:
this is because first is of ratio 1:4 and second one is 2:1 as both pools are used for swap. therefore amount should be amount/8 adding a descriptive comment